internal/csi-addons/rbd/replication.go

```diff
@@ -937,3 +947,35 @@ func checkVolumeResyncStatus(ctx context.Context, localStatus types.SiteStatus)
 
 	return nil
 }
+
+func getCurrentReplicationStatus(mirrorGlobalStatus types.GlobalStatus,
+) (replication.GetVolumeReplicationInfoResponse_Status, string, error) {
+	// get image local status
+	localStatus, err := mirrorGlobalStatus.GetLocalSiteStatus()
+	if err != nil {
+		return replication.GetVolumeReplicationInfoResponse_UNKNOWN, "", err
+	}
+
+	// parse only the message from the description of the image/group status
+	statusDesc := localStatus.GetDescription()
+	desc := strings.SplitN(statusDesc, ",", 2)[0]
+
+	// check if mirroring state is up/down
+	isUp := localStatus.IsUP()
+
+	if !isUp {
+		return replication.GetVolumeReplicationInfoResponse_DEGRADED, desc, nil
+	}
+
+	resp := replication.GetVolumeReplicationInfoResponse_UNKNOWN
+	switch localStatus.GetState() {
+	case librbd.MirrorImageStatusStateError.String():
+		resp = replication.GetVolumeReplicationInfoResponse_ERROR
+	case librbd.MirrorImageStatusStateReplaying.String():
+		fallthrough
```

Comment:
instead of fallthrough add mutliplecase to avoid problems in future? where we need to add more states?

Reply:
add a comment if you still need to keep fallthrough, its easy to make mistakes after few years it will go unnoticed if we add another case b/w.

Reply:
Had to remove fallthrough as golangci-lint was complaining, but added a comment for future ref.